lms/djangoapps/certificates/api.py

```diff
@@ -283,24 +284,21 @@ def set_cert_generation_enabled(course_key, is_enabled):
         log.info("Disabled self-generated certificates for course '%s'.", str(course_key))
 
 
-def is_certificate_invalid(student, course_key):
-    """Check that whether the student in the course has been invalidated
-    for receiving certificates.
+def is_certificate_invalidated(student, course_key):
```

Comment:
The GeneratedCertificate model provides a `is_valid()` method that is a totally different concept from _invalidating_ a certificate. I opted to to use this for my "can be added to allowlist" check as it already existed and I didn't want to duplicate code.

I opted for updating the name of the function to hopefully cause others less confusion in the future...